src/odemis/acq/align/autofocus.py

```diff
@@ -520,8 +525,9 @@ def _DoExhaustiveFocus(future, detector, emt, focus, dfbkg, good_focus, rng_focu
 
         # if nothing was found return to original position and start going
         # downwards
-        focus.moveAbsSync({"z": orig_pos})
-        for next_pos in numpy.arange(orig_pos - step, lower_bound, -step):
+        focus.moveAbsSync({"z": orig_pos})  # TODO does this do anything?
```

Comment:
I think so... it goes back to the original position? Or is it already supposed to be there?

Reply:
Yeah it goes back to the original position but does not do any measurements at that position. It moves to `next_pos` on line 552 before measuring anything, or if it does not go into the for loop it moves to `best_pos` on line 570.

Reply:
Indeed... making unnecessary moves is not a great idea, so I'd suggest to drop that line.